### PR TITLE
Fix the length change in hex mode to change OTP length

### PR DESCRIPTION
**Pull request by ncthompson**

### CODE REVIEW

File: otp.go [hunk 1]
```diff
@@ -82,9 +82,13 @@ func (o *OTP) generateOTP(input int) string {
 		((int(hmacHash[offset+2] & 0xff)) << 8) |
 		(int(hmacHash[offset+3]) & 0xff)
 
-	if o.format == FormatDec {
+	switch o.format {
+	case FormatDec:
 		code = code % int(math.Pow10(o.digits))
+	case FormatHex:
+		code = code >> (32 - 4 * uint(o.digits))
```
[the remark this answers is not in the document]
Comment (ncthompson) [reply otp.go]: Good point, I think even in the Decimal case this will break things. I will add a length check.

File: otp.go [hunk 2]
```diff
@@ -33,7 +33,18 @@ const (
 	FormatHex
 )
 
-func NewOTP(secret string, digits int, hasher *Hasher, format Format) OTP {
+// MaxOTPLength set the character length limit of the library
+const MaxOTPLength = 8
+
+func newOTP(secret string, digits int, hasher *Hasher, format Format) (*OTP, error) {
```
Comment (jedrivisser) [inline otp.go]: Is this supposed to become private?

Comment (jedrivisser) [inline otp.go]: Is this supposed to become private?

Comment (ncthompson) [reply otp.go]: Yes, the `newOTP` function is never called directly. It is also not useful to call it directly as non of the functions are public. It made more sense to change it to a private function.

**Pull request merged**